ext/opentelemetry-ext-docker-tests/tests/mysql/test_mysql_functional.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2020, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import time
+import unittest
+
+import mysql.connector
+
+from opentelemetry import trace as trace_api
+from opentelemetry.ext.mysql import trace_integration
+from opentelemetry.sdk.trace import Tracer, TracerProvider
+from opentelemetry.sdk.trace.export import SimpleExportSpanProcessor
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import (
+    InMemorySpanExporter,
+)
+from pytest import fixture
+
+MYSQL_USER = os.getenv("MYSQL_USER ", "testuser")
+MYSQL_PASSWORD = os.getenv("MYSQL_PASSWORD ", "testpassword")
+MYSQL_HOST = os.getenv("MYSQL_HOST ", "localhost")
+MYSQL_PORT = int(os.getenv("MYSQL_PORT ", "3306"))
+MYSQL_DB_NAME = os.getenv("MYSQL_DB_NAME ", "opentelemetry-tests")
+
+
+class TestFunctionalMysql(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls._connection = None
+        cls._cursor = None
+        cls._tracer_provider = TracerProvider()
+        cls._tracer = Tracer(cls._tracer_provider, None)
+        cls._span_exporter = InMemorySpanExporter()
+        cls._span_processor = SimpleExportSpanProcessor(cls._span_exporter)
+        cls._tracer_provider.add_span_processor(cls._span_processor)
+        trace_integration(cls._tracer)
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls._connection:
+            cls._connection.close()
+
+    def setUp(self):
+        self._span_exporter.clear()
+
+    @fixture(autouse=True)
```

Comment:
IIRC unittest setup/teardown and pytest fixtures don't play well together. Would this do the same thing in `setupClass` or do you need this to be a fixture?

I dislike autouse fixtures in general because using them means requiring pytest to make tests work, they're highly ~magical~ and so a hassle to debug. But since this is only used in this module and these tests really do need a connected DB to run, I don't have much justification for complaints here.

Reply:
I see @toumorokoshi left the same comment in https://github.com/open-telemetry/opentelemetry-python/pull/528/files#r401712822. I'll reply there.